src/service.rs

```diff
@@ -984,8 +999,8 @@ where
                     },
                 ));
                 self.distribute_to_session();
-                return;
             }
+            return;
```

Comment:
Why move `return` here? Another bug?

Reply:
yes, it should be returned immediately, whether send an event on or not.